src/engine.rs

```diff
@@ -670,6 +670,16 @@ impl Reedline {
                 // A handled Event causes a repaint
                 Ok(EventStatus::Handled)
             }
+            ReedlineEvent::DeleteHistoryItem => {
+                match self
+                    .history_cursor
+                    .delete_item_at_cursor(self.history.as_mut())
```

Comment:
Do we need to update what is visible in the traditional `Ctrl-r` search? (this is not the menu but the `reedline::ReedlineEvent::SearchHistory` triggered search.)

Reply:
If you mean `self.update-buffer_from_history()`, it doesn't seem like it, as `history_search_paint()` is used instead of `buffer_paint()`, and that function doesn't support `SubstringSearch`.